packages/core/src/routes/sign-in-experience.ts

```diff
@@ -41,11 +52,48 @@ export default function signInExperiencesRoutes<T extends AuthedRouter>(router:
       body: SignInExperiences.createGuard.omit({ id: true }).partial(),
     }),
     async (ctx, next) => {
-      const { body } = ctx.guard;
+      const { socialSignInConnectorIds, ...rest } = ctx.guard.body;
+      const { signInMethods } = rest;
+
+      if (signInMethods) {
+        const connectorInstances = await getConnectorInstances();
+
+        if (signInMethods.email !== SignInMethodState.disabled) {
+          checkEnabledConnectorByType(connectorInstances, ConnectorType.Email);
```

Comment:
can we replace this function with an `assertThat`?

Reply:
refactored, PTAL
```ts
const assertEnabledConnectorByType = (
  connectorInstances: ConnectorInstance[],
  type: ConnectorType
) => {
  assertThat(
    connectorInstances.some((item) => item.connector.enabled && item.metadata.type === type),
    'sign_in_experiences.enabled_connector_not_found',
    { type }
  );
};
```

Reply:
also replaced other `if(){ throw }` with `assertThat`, PTAL